python/lsst/daf/butler/_butler.py

```diff
@@ -1243,13 +1250,19 @@ def getDirect(self, ref: DatasetRef, *, parameters: Optional[Dict[str, Any]] = N
         parameters : `dict`
             Additional StorageClass-defined options to control reading,
             typically used to efficiently read only a subset of the dataset.
+        storageClass : `StorageClass` or `str`, optional
+            The storage class to be used to override the Python type
+            returned by this method. By default the returned type matches
+            the dataset type definition for this dataset. Specifying a
+            read `StorageClass` can force a different type to be returned.
+            This type must be compatible with the original type.
 
         Returns
         -------
         obj : `object`
             The dataset.
         """
-        return self.datastore.get(ref, parameters=parameters)
+        return self.datastore.get(ref, parameters=parameters, storageClass=storageClass)
 
     def getDirectDeferred(
```

Comment:
I had forgotten about the deferred versions. Given that this takes a parameters parameter, I imagine we would like the ability to force a new storage class to be used later for the get...

Reply:
Up to you.  In the primary `PipelineTask` use case for `DeferredDatasetHandle`, I'd expect the storage class override to be declared in the connections so we wouldn't need that.

Reply:
I added it, although I decided not to add it to the InMemoryDatasetHandle since it seemed a bit odd to create one of those with a python object and immediately declare you wanted it to be returned as a completely different type.